frontend/packages/console-plugin-sdk/README.md

```diff
@@ -0,0 +1,234 @@
+Every (non-core) plugin package must have an `OWNERS` file. Its directory name should end with `-plugin` to denote its purpose.
+
+## Package.json
+
+Plugins are packages. The package definition of a plugin should include a `consolePlugin`
+map that defines the plugin entry module (value of `entry` key), and the plugin integration tests.
+
+The `consolePlugin.entry` path should point to a module that exports the Plugin object (plugin entry module).
+
+From plugin author perspective, a Plugin is simply a list of extensions.
+The plugin entry module is loaded right upon Console startup, so it should lazy-load other code (use the import function) instead of directly referring to it.
+
+``` json
+// File: package.json
+
+{
+  "name": "@console/demo-plugin",
+  "version": "0.0.0-fixed",
+  "description": "Demo plugin for Console web application",
+  "private": true,
+  "dependencies": {
+    "@console/plugin-sdk": "0.0.0-fixed",
+    "@console/shared": "0.0.0-fixed"
+  },
+  "devDependencies": {
+    "@console/internal-integration-tests": "0.0.0-fixed"
+  },
+  "consolePlugin": {
+    "entry": "src/plugin.tsx",
+    "integrationTests": {
+      "demo": [
+        "@console/internal-integration-tests/tests/base.scenario.ts",
+        "integration-tests/tests/demo.scenario.ts"
+      ]
+    }
+  }
+}
+```
+
+## Plugin Entry Module (`.../src/plugin.tsx`)
+
+The Plugin Entry Module is defined in `package.json` in the `consolePlugin.entry` variable,
+relative to the plugin root directory.
+
+``` json
+// File: package.json
+
+{
+  ...
+  "consolePlugin": {
+    "entry": "src/plugin.tsx",
+    ...
+  }
+}
+```
+
+The Plugin Entry Module will export an array of plugin objects of type `Plugin<ConsumedExtensions>`. Each `Plugin` object have a type
+declaring the plugin capabilities and options (for example `Page/Route`, `Dashboards/Card` types).
+
+``` es6
+// File: src/plugin.tsx
+
+const plugin: Plugin<ConsumedExtensions> = [
+  ...
+  {
+    type: 'Dashboards/Tab',
+    properties: {
+      id: 'persistent-storage',
+      title: 'Persistent Storage',
+    },
+  },
+  ...
+];
+
+export default plugin;
+```
+
+See the [demo plugin](frontend/packages/console-demo-plugin) for available Plugin extension points.
+
+For better type checking and code completion, use a type parameter that represents the union of all the extension types consumed by the plugin.
+
+``` es6
+// don't do this
+const plugin: Plugin<any> = [ /* stuff */ ];
+// do this
+const plugin: Plugin<FooExtension | BarExtension> = [ /* stuff */ ];
+// or better yet, this
+type ConsumedExtensions = FooExtension | BarExtension;
+const plugin: Plugin<ConsumedExtensions> = [ /* stuff */ ];
+```
+
+## Monorepo Architecture
+
+Console plugins are 1st class citizens of Console monorepo and therefore reuse the same testing (Jest, Protractor, etc.) infrastructure and conventions.
```

Comment:
"Console plugins are 1st class citizens of Console monorepo"

"Console plugins are first class citizens of the Console monorepo"